### PR TITLE
chore(sdk/testing): add support for `axe-core` version `4.8`

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2700502](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2700502) [AB#2814566](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2814566)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1983/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1983/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1983/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1983/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1983/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1983/playground/)

File: libs/sdk/testing/src/lib/matchers/matchers.spec.ts [hunk 1]
```diff
@@ -33,12 +33,20 @@ function createPassingElement(): any {
...
-  });
+  element.setAttribute('role', 'directory');
+
+  if (params?.includeRelatedNodesError) {
+    // Related nodes include additional reporting.
+    const elem1 = element.firstElementChild as HTMLElement;
+    elem1.setAttribute('role', 'grid');
+    elem1.appendChild(document.createElement('textarea'));
+  }
+
```
Comment (johnhwhite) [inline libs/sdk/testing/src/lib/matchers/matchers.spec.ts:49]: Changed b/c `duplicate-id` is no longer a failing error.

Comment (Blackbaud-SteveBrush) [reply libs/sdk/testing/src/lib/matchers/matchers.spec.ts:49]: Why not? 😅 Seems like it should still fail an accessibility check. Is our test not setup correctly to trigger the failure or did `axe-core` remove that as a rule?

Comment (johnhwhite) [reply libs/sdk/testing/src/lib/matchers/matchers.spec.ts:49]: `axe-core` downgraded `duplicate-id` to `needs-review` rather than `failure`.

Comment (Blackbaud-SteveBrush) [reply libs/sdk/testing/src/lib/matchers/matchers.spec.ts:49]: It looks like they refined the error and created a new one, `duplicate-id-active` https://dequeuniversity.com/rules/axe/4.8/duplicate-id-active. The difference between the two rules is whether the ID is repeated on focusable elements.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**